src/SessionManager/WebBrowser.cc

```diff
@@ -0,0 +1,135 @@
+/* This file is part of the CARTA Image Viewer: https://github.com/CARTAvis/carta-backend
+   Copyright 2018, 2019, 2020, 2021 Academia Sinica Institute of Astronomy and Astrophysics (ASIAA),
+   Associated Universities, Inc. (AUI) and the Inter-University Institute for Data Intensive Astronomy (IDIA)
+   SPDX-License-Identifier: GPL-3.0-or-later
+*/
+
+#include "WebBrowser.h"
+#include "Logger/Logger.h"
+#include "Util.h"
+
+#include <signal.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <filesystem>
+#include <queue>
+#include <sstream>
+#include <stdexcept>
+
+namespace fs = std::filesystem;
+
+namespace carta {
+
+WebBrowser::WebBrowser(const std::string& url, const std::string& browser_cmd) {
+    _url = url;
+    if (browser_cmd.size() > 0) {
+        _cmd = browser_cmd;
+        ParseCmd();
+    }
+    if (_cmd.size() > 0 && _path_exists) {
+        spdlog::debug("WebBrowser: custom command is {}, attempting to open the browser now.", _cmd);
+        OpenBrowser();
+    } else {
+        spdlog::debug("WebBrowser: using default browser.");
+        OpenSystemBrowser();
+    }
+}
+
+void WebBrowser::ParseCmd() {
+    if (_cmd[_cmd.size() - 1] == '&') {
```

Comment:
do we still need to do this?

Reply:
> do we still need to do this?

I don't think it's fully needed, but I also think it isn't of any harm. The new process is launched and detached from the parent process, and output is ignored, so there is no need to send it to the background. So I am open to keeping it or leaving it, not sure if it can be harmful or not.